alma/lib/Utils/CustomerHelper.php

```diff
@@ -0,0 +1,199 @@
+<?php
+/**
+ * 2018-2022 Alma SAS
+ *
+ * THE MIT LICENSE
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+ * Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+ * WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ *
+ * @author    Alma SAS <contact@getalma.eu>
+ * @copyright 2018-2022 Alma SAS
+ * @license   https://opensource.org/licenses/MIT The MIT License
+ */
+
+namespace Alma\PrestaShop\Utils;
+
+use Alma\PrestaShop\Model\AddressData;
+use Alma\PrestaShop\Model\CarrierData;
+use Alma\PrestaShop\Model\CartData;
+use Cart;
+use Context;
+use Country;
+use Customer;
+use Order;
+
+if (!defined('_PS_VERSION_')) {
+    exit;
+}
+
+/**
+ * Class CustomerHelper.
+ *
+ * Customer Helper
+ */
+class CustomerHelper
+{
+    public function __construct(
```

Comment:
customer, context and cart are not declared as class properties

Reply:
You have now to add `addressData`, `customerData`, `carrierData` and `customerOrderHelper` as properties...

Please, ensure you really use `customer`, `context` and `cart` intenally in the class, otherwise, it is not necessary to store them in class properties